### PR TITLE
Pseergeev/streaming tools overview report

### PR DESCRIPTION
**Pull request by mark-rom**
### Jira Ticket
https://luxproject.luxoft.com/jira/browse/STVCIS-4172

### Purpose
build compare report for streaming tools

### Technical steps
- [x] Step 1
- [x] Step 2
- [ ] Step 3

### :octocat: Related PR'S
https://github.com/luxteam/rpr_pipelines/pull/1846
https://github.com/luxteam/jobs_launcher/pull/459
### Jenkins Builds
Tool overview report
https://rpr.cis.luxoft.com/job/DevJobs/job/StreamingSDKComparison/84/
Game overview report

### CODE REVIEW

**Review by Speedwag00n — CHANGES_REQUESTED** [review]

File: drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/streaming_actions.py [hunk 1]
```diff
@@ -301,7 +307,8 @@ def close_sunshine(args, case, process):
         main_logger.info("Start closing")
 
         if args.execution_type != "server":
-            psutil.Popen(["taskkill", "/f", "/im", "moonlight.exe"])
+            MoonlightMixin.close_moonlight()
+            # psutil.Popen(["taskkill", "/f", "/im", "moonlight.exe"])
```
[the remark this answers is not in the document]
Comment (mark-rom) [reply drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/streaming_actions.py]: done

File: drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/client_actions.py [hunk 2]
```diff
@@ -487,16 +492,28 @@ def parse(self):
...
+        self.wait_server_answer(analyze_answer=True, abort_if_fail=True)
 
         sleep(3)
 
         pyautogui.press("S")
 
+    def activate_ssdk_window(self):
+        params = self.params.copy()
+        for action in (Move, DoSleep):
+            params["arguments_line"] = "1000 800" if action.__name__ == "Move" else "1"
```
[the remark this answers is not in the document]
Comment (mark-rom) [reply drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/client_actions.py]: done

File: drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/utils.py [hunk 3]
```diff
@@ -94,14 +95,18 @@ def collect_traces(archive_path, archive_name):
...
         except Exception:
             pass
 
         script = "powershell \"Start-Process cmd '/k cd \"{}\" && .\\log_extended.cmd & exit 0' -Verb RunAs\"".format(gpuview_path)
 
+        start = datetime.now()
         proc = psutil.Popen(script, stdout=PIPE, stderr=PIPE, shell=True)
+        end = datetime.now()
 
+        main_logger.debug(f"total execution time of the script {(end - start).total_seconds()}")
```
Comment (Speedwag00n) [inline drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/utils.py:109]: Удали эти изменения

**Review by Speedwag00n — CHANGES_REQUESTED** [review]

File: drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/client_actions.py [hunk 4]
```diff
@@ -487,16 +498,30 @@ def parse(self):
             self.keys = ""
 
     def execute(self):
+        if self.args.streaming_type in (StreamingType.SDK, StreamingType.FULL_SAMPLES):
+            # for correct work of Streaming SDK and Streaming FS
+            # we need to click on the streaming window
+            self.activate_ssdk_window()
```
[the remark this answers is not in the document]
Comment (mark-rom) [reply drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/client_actions.py:504]: уже удалена

File: drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/utils.py (hunk 3 above)
Comment (mark-rom) [reply drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/utils.py]: done

File: drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/server_actions.py [hunk 5]
```diff
@@ -586,6 +587,9 @@ def execute(self):
                     if is_simultaneous_connection(self.case):
                         sleep(5)
 
+                if self.args.game_name == "LatencyTool":
+                    sleep(40)
```
[the remark this answers is not in the document]
Comment (mark-rom) [reply drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/server_actions.py]: done

File: drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/server_actions.py [hunk 6]
```diff
@@ -577,7 +578,7 @@ def execute(self):
                         sleep(5)
 
             if self.process is None:
-                should_collect_traces = (self.args.collect_traces == "BeforeTests")
+                should_collect_traces = (self.args.collect_traces == "AfterTests" and self.args.streaming_type == StreamingType.SDK)
```
[the remark this answers is not in the document]
Comment (mark-rom) [reply drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/server_actions.py]: done

File: drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/server_actions.py [hunk 7]
```diff
@@ -234,7 +234,8 @@ def analyze_result(self):
 class NextCase(Action):
     @Action.server_action_decorator
     def execute(self):
-        if self.params["args"].track_used_memory:
+        if self.params["args"].track_used_memory == "True":
+            self.logger.debug(f"track used memory is type {type(self.params['args'].track_used_memory)} and is {self.params['args'].track_used_memory}")
```
[the remark this answers is not in the document]
Comment (mark-rom) [reply drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/server_actions.py]: done

File: drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/rules/rules_processor.py [hunk 8]
```diff
@@ -71,7 +71,8 @@ def __init__(self, args, case, json_content, log_path, log_type):
         self.highlighter = LogsHighLighter(self.log_path)
 
         for rule_name in GENERAL_RULES:
-            if "PRSC" in case["case"] or "SUN" in case["case"]:
+            if "PRSC" in case["case"] or "SUN" in case["case"] or "GMS" in case["case"]:
+                main_logger.debug("General rules don't apply for Gamestream")
```
[the remark this answers is not in the document]
Comment (mark-rom) [reply drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/rules/rules_processor.py]: done

File: drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/client_actions.py [hunk 9]
```diff
@@ -448,7 +459,7 @@ def execute(self):
             # start server before client
             if "start_first" in self.case and self.case["start_first"] == "server":
                 if self.process is None:
-                    should_collect_traces = (self.args.collect_traces == "BeforeTests")
+                    should_collect_traces = (self.args.collect_traces == "AfterTests" and self.args.streaming_type == StreamingType.SDK)
```
[the remark this answers is not in the document]
Comment (mark-rom) [reply drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/client_actions.py]: done

File: drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/client_actions.py [hunk 10]
```diff
@@ -434,12 +434,23 @@ def execute(self):
...
                         if is_simultaneous_connection(self.case):
                             sleep(5)
 
+            # add logic to unfold parsec window if latancy
+            if self.args.streaming_type == StreamingType.PARSEC and self.args.game_name == "LatencyTool": # noqa
+                hwnd = win32gui.FindWindow(None, "Parsec")
+                make_window_active(hwnd)
+            # elif self.args.streaming_type == (StreamingType.SUNSHINE, StreamingType.GAMESTREAM) and self.args.game_name == "LatencyTool": # noqa
+            #     hwnd = win32gui.FindWindow(None, "Moonlight")
+            #     make_window_active(hwnd)
```
[the remark this answers is not in the document]
Comment (mark-rom) [reply drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/client_actions.py]: done

File: drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/client_actions.py [hunk 11]
```diff
@@ -434,12 +434,23 @@ def execute(self):
...
+                    if self.args.game_name == "LatencyTool":
+                        sleep(40)
+
                     if should_collect_traces:
                         collect_traces(self.archive_path, self.archive_name + "_client.zip")
                     elif "start_first" in self.case and self.case["start_first"] == "client":
                         if is_simultaneous_connection(self.case):
                             sleep(5)
 
+            # add logic to unfold parsec window if latancy
```
[the remark this answers is not in the document]
Comment (mark-rom) [reply drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/client_actions.py]: done

File: drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/client_actions.py [hunk 12]
```diff
@@ -434,12 +434,23 @@ def execute(self):
                         # call start_streaming function just to change settings
                         start_streaming(self.args, self.case, script_path=None, socket=self.sock)
 
+                    if self.args.game_name == "LatencyTool":
+                        sleep(40)
```
[the remark this answers is not in the document]
Comment (mark-rom) [reply drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/client_actions.py]: done

File: drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/client_actions.py [hunk 13]
```diff
@@ -418,13 +418,13 @@ def execute(self):
...
-            if self.args.streaming_type in (StreamingType.FULL_SAMPLES, StreamingType.PARSEC, StreamingType.SUNSHINE):
+            if self.args.streaming_type in (StreamingType.FULL_SAMPLES, StreamingType.PARSEC, StreamingType.SUNSHINE, StreamingType.GAMESTREAM):
                 self.sock.send(self.action.encode("utf-8"))
 
             # start client before server (default case)
             if "start_first" not in self.case or self.case["start_first"] != "server":
-                if self.process is None or self.args.streaming_type in (StreamingType.FULL_SAMPLES, StreamingType.PARSEC, StreamingType.SUNSHINE):
-                    should_collect_traces = (self.args.collect_traces == "BeforeTests")
+                if self.process is None or self.args.streaming_type in (StreamingType.FULL_SAMPLES, StreamingType.PARSEC, StreamingType.SUNSHINE, StreamingType.GAMESTREAM):
+                    should_collect_traces = (self.args.collect_traces == "AfterTests" and self.args.streaming_type == StreamingType.SDK)
```
[the remark this answers is not in the document]
Comment (mark-rom) [reply drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/client_actions.py]: done

File: drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/client_actions.py [hunk 14]
```diff
@@ -409,7 +409,7 @@ def execute(self):
         if self.args.streaming_type == StreamingType.AMD_LINK:
             self.sock.send(self.action.encode("utf-8"))
 
-            should_collect_traces = (self.args.collect_traces == "BeforeTests")
+            should_collect_traces = (self.args.collect_traces == "AfterTests" and self.args.streaming_type == StreamingType.SDK)
```
[the remark this answers is not in the document]
Comment (mark-rom) [reply drivers/amf/stable/tools/tests/StreamingSDKTests/jobs/Scripts/client_actions.py]: done

**Review by Speedwag00n — APPROVED** [review]

**Pull request merged**